hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MutableRollingAverages.java

```diff
@@ -167,7 +167,7 @@ synchronized void replaceScheduledTask(int windows, long interval,
   }
 
   @Override
-  public void snapshot(MetricsRecordBuilder builder, boolean all) {
+  public synchronized void snapshot(MetricsRecordBuilder builder, boolean all) {
```

Comment:
Do we need synchronized here? Its super class is not.

Reply:
I have the same question.

Reply:
@ferhui @tomscut Thanks your comment.


SpotBugs | module:hadoop-common-project/hadoop-common
-- | --
  | Inconsistent synchronization of org.apache.hadoop.metrics2.lib.MutableRollingAverages.recordValidityMs; locked 66% of time Unsynchronized access at MutableRollingAverages.java:66% of time Unsynchronized access at MutableRollingAverages.java:[line 182]

So consider add synchronized.
Reference code MutableRatesWithAggregation#snapshot, the  method to add the synchronized

> public class MutableRatesWithAggregation extends MutableMetric {
> @Override
>   public synchronized void snapshot(MetricsRecordBuilder rb, boolean all) {
>     Iterator<WeakReference<ConcurrentMap<String, ThreadSafeSampleStat>>> iter =
>         weakReferenceQueue.iterator();
>     ...
>   }
> }

Reply:
I'm not sure  whether the synchronized is needed in here, the rest of part look good to me.

Reply:
@ferhui @tomscut @AlphaGouGe Thanks your comment.
this logic is only called when DataNode JMX get SendPacketDownstreamAvgInfo Metrics, 
so there is no concurrency security issue at the moment, no need to add synchronized method.
Update PR.